### PR TITLE
[Feature-13428][Resource Center] Add support for Google Cloud Storage in the resource center

### LINKED ISSUE #13428: [Feature][Resource Center] Add support for Google Cloud Storage in the resource center
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Add support for [Google Cloud Storage](https://cloud.google.com/storage) in the resource center, like OSS and S3.

### Use case

_No response_

### Related issues

* #11708

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by rickchengx**
## Purpose of the pull request

* close: #13428 

## Brief change log

* Add support for writing task logs to Google Cloud Storage

## Verify this pull request

UT and manually tested
1. configure in the `common.properties`
```
resource.storage.type=GCS
# the location of the google cloud credential, required if you set resource.storage.type=GCS
resource.google.cloud.storage.credential=/path/to/credential
# gcs bucket name, required if you set resource.storage.type=GCS
resource.google.cloud.storage.bucket.name=<your-bucket>
```
2. Upload files and directory in the resource center
截屏2023-02-08 15 11 10

3. Check files in the GCS
截屏2023-02-08 15 10 54

### CODE REVIEW

**Comment by rickchengx** [comment]
I will add related LICENSEs later.

**Comment by rickchengx** [comment]
Hi, @zhongjiajie , @SbloodyS , @EricGao888 ,could you please help review this?

File: dolphinscheduler-storage-plugin/dolphinscheduler-storage-gcs/src/main/java/org/apache/dolphinscheduler/plugin/storage/gcs/GcsStorageOperator.java [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+        blob.downloadTo(Paths.get(dstFilePath));
+    }
+
+    @Override
+    public boolean exists(String fullName) throws IOException {
+        return isObjectExists(fullName);
+    }
+
+    @Override
+    public boolean delete(String filePath, boolean recursive) throws IOException {
```
Comment (zhongjiajie) [inline dolphinscheduler-storage-plugin/dolphinscheduler-storage-gcs/src/main/java/org/apache/dolphinscheduler/plugin/storage/gcs/GcsStorageOperator.java:165]: it seems the parameter `recursive` not use in this method, so GCS does not support delete recursive in this method?

Comment (rickchengx) [reply dolphinscheduler-storage-plugin/dolphinscheduler-storage-gcs/src/main/java/org/apache/dolphinscheduler/plugin/storage/gcs/GcsStorageOperator.java:165]: Hi, @zhongjiajie , thanks a lot for your kind review. Perhaps for some historical reasons, `recursive` is not used (also in the `S3StorageOperator` and `OssStorageOperator`).

Another interface below is called when deleting with a `childrenPathList`. 

```
@Override
    public boolean delete(String fullName, List<String> childrenPathList, boolean recursive) throws IOException {
        // append the resource fullName to the list for deletion.
        childrenPathList.add(fullName);

        boolean result = true;
        for (String filePath : childrenPathList) {
            if (!delete(filePath, recursive)) {
                result = false;
            }
        }

        return result;
    }
```

**Review by zhongjiajie — APPROVED** [review]

**Pull request merged**